src/main/kotlin/de/achimonline/ansible_lint/common/AnsibleLintTempEnv.kt

```diff
@@ -30,7 +30,10 @@ class AnsibleLintTempEnv(
     val symlinks: List<Path>
 
     init {
-        val relativeFilePath = fileToLint.virtualFile.path.removePrefix(projectBasePath)
+        val normalizedProjectBasePath = Paths.get(projectBasePath).normalize().toString();
+        val normalizedFileToLint = Paths.get(fileToLint.virtualFile.path).normalize().toString();
```

Comment:
[normalize()](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/nio/file/Path.html#normalize()) via `Path` might work here, but its main focus is not to convert Windows-file-separators to Unix-file-separators.

Using [toNioPath()](https://github.com/JetBrains/intellij-community/blob/acf7d84b3f6fbe95f80462b2862b4b4e6fe358eb/platform/core-api/src/com/intellij/openapi/vfs/VirtualFile.java#L139-L167) from `VirtualFile` is a better choice.

BTW... this is Kotlin. No semicolon needed at the end of line.

Reply:
> [normalize()](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/nio/file/Path.html#normalize()) via Path might work here, but its main focus is not to convert Windows-file-separators to Unix-file-separators.

This actually works in backwards of what you said - the normalize() here converts URI-like / separators to (OS specific separators) in this example Windows backslashes \\. Since only the virtual file system uses URI like paths (with slashes) and everything else uses OS directory separator (thus on Windows backslashes). The normalize() normalizes whatever to the OS specific separators, that's the key here since you are using the OS separator (File.separator) to split the path and other string-based Path operations.

The new implementation looks worse since you aren't using Path.absolutePathString(). Doesn't this open the same door for the Path to return some unexpected results like relative path? I guess the Path.pathString may work in this context but from the documentation stand point this implementation looks fragile the same way as the original implementation.

I don't like to manipulate paths like strings, since that just causes trouble (not only because of directory separators but also because relative paths) but if you need/want to then you should make sure all the paths are absolute and if you work with different file systems (like the Jetbrains VFS) then it's critical to make sure the separators are normalized.

BTW: I didn't write single line of Kotlin in my life (to this very day) but you asked for pull request so that's what you get... 😄 I don't think you would do better - if you try to work your whole life in C/C++/Java/PHP and then switch to Kotlin for 3 minutes... You can't rewire your brain like this... IntelliJ also doesn't highlight this as it does with Python, so it doesn't look like semicolon in Kotlin is a sin.

Reply:
> This actually works in backwards of what you said - the normalize() here converts URI-like / separators to (OS specific separators) in this example Windows backslashes \. Since only the virtual file system uses URI like paths (with slashes) and everything else uses OS directory separator (thus on Windows backslashes). The normalize() normalizes whatever to the OS specific separators, that's the key here since you are using the OS separator (File.separator) to split the path and other string-based Path operations.

Did you even read the official documentation linked above?

`normalize()`:

```
Returns a path that is this path with redundant name elements eliminated.

The precise definition of this method is implementation dependent but in general it derives from this path, a path that does not contain redundant name elements. In many file systems, the "." and ".." are special names used to indicate the current directory and parent directory. In such file systems all occurrences of "." are considered redundant. If a ".." is preceded by a non-".." name then both names are considered redundant (the process to identify such names is repeated until it is no longer applicable).

This method does not access the file system; the path may not locate a file that exists. Eliminating ".." and a preceding name from a path may result in the path that locates a different file than the original path. This can arise when the preceding name is a symbolic link.
```

There is no mention about backslashes or separators at all.
The focus here is removing _redundant_ parts (mainly `.` and `..`).

And that is what I said.

Whereas `to(Nio)Path()` clearly says:

```
   * Gets the path of this file. Path is a string that uniquely identifies a file within a given
   * {@link VirtualFileSystem}. Format of the path depends on the concrete file system.
   * For {@link LocalFileSystem} it is an absolute file path with file separator characters
   * ({@link File#separatorChar}) replaced to the forward slash ({@code '/'}). If you need to show path in UI, use {@link #getPresentableUrl()}
   * instead.
```

The conversion of path-separators is explicitly mentioned here.

> The new implementation looks worse since you aren't using Path.absolutePathString(). Doesn't this open the same door for the Path to return some unexpected results like relative path? I guess the Path.pathString may work in this context but from the documentation stand point this implementation looks fragile the same way as the original implementation.

Well... I honestly don't care what (you in particular) "think".
And I'd appreciate if you'd move along and find some place else to place your "clever analysis"  statements.

Thank you.

Reply:
> The focus here is removing redundant parts (mainly . and ..).

Yes that's very important point. Since /some/another won't match /some/./another or /some/third/../another in string yet it's the same path. That's exactly what you want.

> The conversion of path-separators is explicitly mentioned here.

As you mentioned that's not enough. The relative links will break string-based comparison. 

> Well... I honestly don't care what (you in particular) "think".
> And I'd appreciate if you'd move along and find some place else to place your "clever analysis" statements.

It works as I said it's just silly if you want to be pedantic to make another fragile implementation that will fail with relative paths.

Reply:
![image](https://github.com/user-attachments/assets/4affc45b-98f3-49b0-b384-dad03a669675)

The function you linked does exactly the opposite thing you posted it should do.
That's the only reason why it can even work.

If it did function as you posted, it would brake the plugin...